### PR TITLE
getquote

### PR DESCRIPTION
**Pull request by 0xAlec**
**What changed? Why?**

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/token/types.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export type GetTokensOptions = {
  * Note: exported as public Type
  */
 export type Token = {
-  address: Address; // The address of the token contract
+  address: Address | ''; // The address of the token contract
```
Comment (0xAlec) [inline src/token/types.ts:59]: this is since ETH has `''` as the `Address` field

File: src/definitions/swap.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 export const ListSwapAssets = 'cdp_listSwapAssets';
+export const GetSwapQuote = 'cdp_getSwapQuote';
```
Comment (Zizzamia) [inline src/definitions/swap.ts:2]: Side point, we might want to use the UPPER_CASE standard for those const, and keep them alphabetical order. So do.

```
export const CDP_GET_SWAP_QUOTE = 'cdp_getSwapQuote';
export const CDP_LIST_SWAP_ASSETS = 'cdp_listSwapAssets';
```

File: src/token/types.ts (hunk 1 above)
Comment (kyhyco) [reply src/token/types.ts:59]: is this always going to be true?

File: src/swap/types.ts [hunk 3]
```diff
@@ -1,23 +1,68 @@
 import { Address } from 'viem';
-import { Token } from '../token/types';
+import { LegacyTokenData, Token } from '../token/types';
 
 export type AddressOrETH = Address | 'ETH';
 
+/**
+ * Note: exported as public Type
+ */
+export type GetQuoteParams = {
```
Comment (Zizzamia) [inline src/swap/types.ts:9]: For all types, let's add a comment that explain each line, similar to what we did here https://github.com/coinbase/onchainkit/blob/main/src/token/types.ts#L50

And keep all types alphabetical ordered.

File: src/swap/types.ts [hunk 4]
```diff
@@ -1,23 +1,68 @@
...
+  from: Token;
+  to: Token;
+  amount: string;
```
Comment (Zizzamia) [inline src/swap/types.ts:12]: What's the difference between `amount` and `amountReference`?

File: src/swap/core/getQuote.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    amountReference: 'from',
+  };
+
+  params = { ...defaultParams, ...params };
+
+  // Only pass in the address as a parameter to the request
+  // Defaults to ETH if no address is provided
+  const addressParams: GetQuoteParamsWithAddress = {
+    from: params.from.address || 'ETH',
+    to: params.to.address || 'ETH',
```
Comment (Zizzamia) [inline src/swap/core/getQuote.ts:28]: Why the type can be wither an `Address` or a `string`? I guess the API works like this?

File: src/token/types.ts (hunk 1 above)
Comment (Zizzamia) [reply src/token/types.ts:59]: When we say ETH, are we talking about ETH Base or ETH (L1)?

Comment (0xAlec) [reply src/token/types.ts:59]: either, native ETH does not have an address on any network

also this PR isn't ready yet, will be doing some refactoring once i make some upstream changes to the api

Comment (Zizzamia) [reply src/token/types.ts:59]: awesome

**Pull request closed without merging**